presto-spark-base/src/main/java/com/facebook/presto/spark/execution/PrestoSparkRemoteSourceOperator.java

```diff
@@ -74,18 +81,29 @@ public Page getOutput()
             return null;
         }
 
-        SerializedPage serializedPage = null;
+        PageBuilder pageBuilder = new PageBuilder(types);
         synchronized (iterator) {
-            if (iterator.hasNext()) {
-                serializedPage = iterator.next();
+            while (iterator.hasNext() && !pageBuilder.isFull()) {
+                PrestoSparkRow row = iterator.next();
+                SliceInput sliceInput = new BasicSliceInput(wrappedBuffer(row.getBytes(), 0, row.getLength()));
+                pageBuilder.declarePosition();
+                for (int channel = 0; channel < types.size(); channel++) {
+                    BlockBuilder blockBuilder = pageBuilder.getBlockBuilder(channel);
+                    blockBuilder.readPositionFrom(sliceInput);
+                }
+                sliceInput.close();
```

Comment:
looks like this did nothing for `BasicSliceInput` ?

Reply:
Just in case =)